contracts/Escrow.sol

```diff
@@ -0,0 +1,175 @@
+pragma solidity 0.8.9;
+
+import { IEscrow } from "./interfaces/IEscrow.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { IOracle } from "./interfaces/IOracle.sol";
+import { ILoan } from "./interfaces/ILoan.sol";
+
+contract Escrow is IEscrow {
+
+    // the minimum value of the collateral in relation to the outstanding debt e.g. 10% of outstanding debt
+    uint public minimumCollateralRatio;
+
+    // return if have collateral but no debt
+    uint256 MAX_INT = 115792089237316195423570985008687907853269984665640564039457584007913129639935;
+
+    // Stakeholders and contracts used in Escrow
+    address public loan;
+    address public oracle;
+    address public lender;
+    address public borrower;
+    address public arbiter;
+
+    // tracking tokens that were deposited
+    address[] private _tokensUsedAsCollateral;
+
+    // mapping to check uniqueness of tokensUsedAsCollateral
+    mapping(address => bool) private _tokensUsed;
+
+    // tokens used as collateral (must be able to value with oracle)
+    mapping(address => uint) public deposited;
+
+    // collateral tokens that have been used for farming
+    mapping(address => Farm) public farmedTokens;
+
+    constructor(
+        uint _minimumCollateralRatio,
+        address _oracle,
+        address _lender,
+        address _borrower,
+        address _arbiter
+    ) public {
+        minimumCollateralRatio = _minimumCollateralRatio;
+        oracle = _oracle;
+        lender = _lender;
+        borrower = _borrower;
+        arbiter = _arbiter;
+    }
+
+    /*
+    * @dev see IEscrow.sol
+    */
+    function setLoanContract(address _loan) external {
+        require(msg.sender == borrower, "Escrow: only borrower can call");
+        require(loan == address(0), "Escrow: loan contract already set");
+        loan = _loan;
+    }
+
+    /*
+    * @dev updates the cratio according to the collateral value vs loan value
+    * @returns the updated collateral ratio
+    */
+    function _getLatestCollateralRatio() internal returns(uint) {
+        uint debtValue = ILoan(loan).accrueInterest();
+        uint collateralValue = _getCollateralValue();
+        if(collateralValue == 0) return 0;
+        if(debtValue == 0) return MAX_INT;
+
+        return _percent(collateralValue, debtValue, 18);
+    }
+
+    // https://stackoverflow.com/questions/42738640/division-in-ethereum-solidity/42739843#42739843
+    function _percent(uint numerator, uint denominator, uint precision) internal pure returns(uint quotient) {
+        uint _numerator  = numerator * 10 ** (precision + 1);
+        // with rounding of last digit
+        uint _quotient =  ((_numerator / denominator) + 5) / 10;
```

Comment:
Since the number is divided by 10, adding 5 will mean that it is rounded up if >=5, else rounded down. 

e.g. `954 + 5 / 10 = 95` vs `956 + 5 / 10 = 96`